### ISSUE TITLE
Signature is invalid when created with StateOrProvince or more than one Role

### ISSUE THREAD

**Issue by informatik01** [open-eid/digidoc4j#19]
**Disclaimer**
_I have not deeply researched the cause of the following errors, so I am not absolutely sure whether those bugs are in the DigiDoc4j library code or in the [libdigidocpp](https://github.com/open-eid/libdigidocpp) library code. But because I am using DigiDoc4j in the first place I'm reporting them here_

---
I am signing documents using external signing scenario (with the help of [hwcrypto.js](https://github.com/open-eid/hwcrypto.js)). All goes fine until I create signature with the following meta-data:

#### 1) [StateOrProvince](http://open-eid.github.io/digidoc4j/org/digidoc4j/SignatureBuilder.html#withStateOrProvince-java.lang.String-) (_found by Kertu Hiire from [RIA](http://www.ria.ee)_)
So if this info is added to the signature the signing process completes successfully ("_**Signing BDoc successfully completed**_") and BDOC container is also successfully saved to a file. But when opening this container with _DigiDoc3_ klient (version **3.12.1.1450** - latest version as of now) it cannot be opened and the following error is shown:

> BDoc.cpp:456 Failed to parse signature 'META-INF/signatures0.xml'.
> SignatureBES.cpp:300 Failed to parse signature XML: :1:4124 error: element 'StateOrProvince' is not allowed for content model '(City?,StateOrProvince?,PostalCode?,CountryName?)'

![digidoc3_error_message1](https://cloud.githubusercontent.com/assets/11339880/15360503/77cab7f4-1d16-11e6-80bb-024c9d538c23.png)

#### 2) [Roles](http://open-eid.github.io/digidoc4j/org/digidoc4j/SignatureBuilder.html#withRoles-java.lang.String...-)
When adding more that one role to a signature, again all is fine on the DigiDoc4j side: both signing and saving a container to a file. But when opening the container with DigiDoc3 client, although it still  can be opened (unlike the previous case) it is shown that the signature is not valid:
![digidoc3_error_issue2_1](https://cloud.githubusercontent.com/assets/11339880/15360903/295620fc-1d18-11e6-9313-416175eb3a1d.png)
![digidoc3_error_issue2_2](https://cloud.githubusercontent.com/assets/11339880/15360947/50073f56-1d18-11e6-8440-1ab4bf901032.png)
![digidoc3_error_issue2_3](https://cloud.githubusercontent.com/assets/11339880/15360980/6b02e62a-1d18-11e6-9d54-2801bea7d2ac.png)

---
#### Other thoughts
And as you may see in BOTH issue 1 and 2 error messages it is the code in the [**SignatureBES.cpp**](https://github.com/open-eid/libdigidocpp/blob/master/src/SignatureBES.cpp) file that is responsible for generating those errors. For instance, the restriction (_only one role_) is enforced ONLY for _BDoc-BES format_ as is seen the code inside the _SignatureBES.cpp_ file.

But important part is that I am explicitly using either **Time-mark** ([SignatureProfile.LT_TM](http://open-eid.github.io/digidoc4j/org/digidoc4j/SignatureProfile.html#LT_TM)) or **Time-stamp** ([SignatureProfile.TS](http://open-eid.github.io/digidoc4j/org/digidoc4j/SignatureProfile.html#LT)) so why not [SignatureTS.cpp](https://github.com/open-eid/libdigidocpp/blob/master/src/SignatureTS.cpp) or [SignatureTM.cpp](https://github.com/open-eid/libdigidocpp/blob/master/src/SignatureTM.cpp) are used when opening containers created with the related profiles?

In other words I am not quite sure what is the reason for those errors, i.e. was BDOC container constructed incorrectly or DiciDoc3 client does something wrong.

---
Here is a code excerpt that shows how I create _DataToSign_ object that is to be externally signed:
```java
DataToSign dataToSign = SignatureBuilder.
				aSignature(container).
				withSigningCertificate(signingCert).
				withSignatureDigestAlgorithm(DigestAlgorithm.SHA256).
				withSignatureProfile(SignatureProfile.LT).
				withCountry("Estonia").
				withStateOrProvince("Harjumaa").
				withCity("Tallinn").
				withPostalCode("12618").
				withRoles("Developer", "Sportsman").
				buildDataToSign();
```

**Comment by rvillido** [comment]
Thank you for the feedback. 

We will investigate where it should be fixed.

**Comment by rvillido** [comment]
We have investigated the problem and split it into two issues

1. The StateOrProvince issue will be fixed within DigiDoc4j (and perhaps the schema validation rules will be relaxed in the Client application)
https://www.pivotaltracker.com/story/show/120521065

2. Support for multiple roles will be added to DigiDoc Client

Thanks for a detailed report

**Comment by informatik01** [comment]
You're welcome!

**Comment by rvillido** [comment]
This problem has been fixed in the latest 1.0.4 release.

Signatures can have multiple roles. stateOrProvince and other signature parameter fields can be filled. Such containers can be opened with DigiDoc Client without problems.

Thanks for the bug report.
